crates/forge/test_utils/src/runner.rs

```diff
@@ -99,6 +105,33 @@ impl<'a> TestCase {
 
         dir.child("src/lib.cairo").touch().unwrap();
 
+        let snforge_std_path = Utf8PathBuf::from_str("../../snforge_std")
+            .unwrap()
+            .canonicalize_utf8()
+            .unwrap()
+            .to_string()
+            .replace('\\', "/");
+
+        let scarb_toml_path = dir.child("Scarb.toml");
+        scarb_toml_path
+            .write_str(&formatdoc!(
+                r#"
+                [package]
+                name = "test_package"
+                version = "0.1.0"
+    
+                [[target.starknet-contract]]
+                sierra = true
+                casm = true
+    
+                [dependencies]
+                starknet = "2.3.1"
```

Comment:
This will have to be manually updated with every cairo bump which we'll probably forget about. Perhaps this value should be taken from top-level `.tool-versions` file, assuming scarb will have the same versions as starknet.

Reply:
Out of scope + we should stop updating it manually in all the other places anyways, it should work until a new Cairo (and thus Scarb) major is released.

Imagine if we did not bump it in `snforge_std` for 2.3.1. You will have `starknet=2.3.0` specified, which is the same as `starknet=^2.3.0`, so it would still work with both Scarb 2.3.1 and 2.3.0. Now we basically limited our users to use Scarb 2.3.1 only.

Summary: until a new major of Scarb/Cairo is released, updating starknet dependency in Scarb.tomls for tests is not harmful, but useless. Updating it in `snforge_std` is limiting the range of compatible Scarb versions (we should stop doing that).

Reply:
Yes, https://docs.swmansion.com/scarb/docs/reference/specifying-dependencies.html#version-requirements

Reply:
Ideally we'd test it across some range of values, because this approach only ensures forge works with a specific version. Even if, in theory, it should work with multiple.